crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -1130,8 +1130,9 @@ fn build_move_args<E: fmt::Debug, S: StorageView<E>, Mode: ExecutionMode>(
                     let TypeTag::Struct(struct_tag) = type_tag else {
                         invariant_violation!("Struct type make a non struct type tag")
                     };
+                    let type_ = (*struct_tag).into();
                     ValueKind::Object {
-                        type_: MoveObjectType::Other(*struct_tag),
```

Comment:
I discovered we accidentally did exactly what I'm warning against, which is what motivated the PR. I want to make sure the only way to make a `MoveObjectType` from a `StructTag` is by going through the loading/conversion function, which is unfortunately rather hard without hiding the internals of the type. Though I'm open to suggestions.